packages/ember-glimmer/tests/integration/application/engine-test.js

```diff
@@ -572,4 +587,41 @@ moduleFor('Application test: engine rendering', class extends ApplicationTest {
       });
     });
   }
+
+  ['@test query params don\'t have stickiness by default between model'](assert) {
+    assert.expect(1);
+    let tmpl = '{{#link-to "blog.category" 1337}}Category 1337{{/link-to}}';
+    this.setupAppAndRoutableEngine();
+    this.additionalEngineRegistrations(function() {
+      this.register('template:category', compile(tmpl));
+    });
+
+    return this.visit('/blog/category/1?type=news').then(() => {
+      let suffix = '/blog/category/1337';
+      let href = this.element.querySelector('a').href;
+
+      // check if link ends with the suffix
+      assert.ok(this.stringsEndWith(href, suffix));
+    });
+  }
+
+  ['@test query params in customized controllerName have stickiness by default between model'](assert) {
```

Comment:
> How do we handle scenarios where folks have customized controllerName to specify a controller other than the route's controller is handling their QP's? With this change, I think that use case might break?

@rwjblue I added a test on this and you are correct this change will break queryParam in customized `controllerName`.